src/summary.rs

```diff
@@ -258,7 +258,11 @@ impl Expression {
         rhs: Box::new(Expression::new(rhs)),
       },
       Join { lhs, rhs } => Expression::Join {
-        lhs: Box::new(Expression::new(lhs)),
+        lhs: if lhs.is_some() {
+          Some(Box::new(Expression::new(lhs.as_ref().unwrap())))
+        } else {
+          None
+        },
```

Comment:
I tried cleaning using the `if let Some()` syntax here but was getting some odd warnings && hints from rust-analyzer.

<img width="1282" alt="Screen Shot 2022-09-02 at 12 02 03 AM" src="https://user-images.githubusercontent.com/10244162/188056708-e64c6ce6-763d-4c2f-a1c8-0aa3ccf40301.png">

I was able to pass the tests using that syntax but held off on committing that version. Not sure if there is an issue with that LSP implementation or an actual issue with that syntax in this context.

Reply:
The suggestion to use [`Option::map`](https://doc.rust-lang.org/std/option/enum.Option.html#method.map) that hint makes (which comes from [clippy](https://github.com/rust-lang/rust-clippy)) is probably what you want since it's equivalent and more succinct:

```rust
lhs: lhs.as_ref().map(|lhs| Box::new(Expression::new(lhs)))
```

Reply:
Hmm, if this handles the `None` case then I suppose it is more succinct, but I'd also argue more cryptic. I'll test it out, though. And if it works, I am happy to use that, if it is preferred.

Reply:
Yup, that does it. Haven't come across the method on Option before. Was also a bit thrown off because I am more used to thinking of `map` as a method for lists.

Reply:
> I'll test it out, though. And if it works, I am happy to use that, if it is preferred.

I'd say it's very common in Rust code along with the other combinatorial methods on `Option` and `Result` (`and_then`, `or_else`, `map_err`, etc.)

Reply:
I ended up reading through the rest of the Option methods to see if there was anything else I could pair down. Tried using `and_then` in a couple places, but because of the Results being unwrapped with `?` it didn't end up looking like an improvement to me (though I definitely might have missed something).